src/spawn.cpp

```diff
@@ -312,29 +307,27 @@ bool Spawn::spawnMonster(uint32_t spawnId, spawnBlock_t sb, bool startup /* = fa
 bool Spawn::spawnMonster(uint32_t spawnId, MonsterType* mType, const Position& pos, Direction dir,
                          bool startup /*= false*/)
 {
-	std::unique_ptr<Monster> monster_ptr(new Monster(mType));
-	if (!tfs::events::monster::onSpawn(monster_ptr.get(), pos, startup, false)) {
+	auto monster = std::make_shared<Monster>(mType);
```

Comment:
Could this be `const auto`
```suggestion
	const auto monster = std::make_shared<Monster>(mType);
```